ProjFS.Mac/PrjFSLib/PrjFSKextLogDaemon/PrjFSKextLogDaemon.cpp

```diff
@@ -0,0 +1,163 @@
+#include "../../PrjFSKext/public/PrjFSLogClientShared.h"
+#include "../PrjFSUser.hpp"
+#include <iostream>
+#include <OS/log.h>
+#include <IOKit/IOKitLib.h>
+#include <signal.h>
+
+static const char PrjFSKextLogDaemon_OSLogSubsystem[] = "org.vfsforgit.prjfs.PrjFSKextLogDaemon";
+
+static os_log_t s_daemonLogger, s_kextLogger;
+static IONotificationPortRef s_notificationPort;
+
+static void StartLoggingKextMessages(io_connect_t connection, io_service_t service, os_log_t daemonLogger, os_log_t kextLogger);
+static void SetupExitSignalHandler();
+
+int main(int argc, const char* argv[])
+{
+    s_daemonLogger = os_log_create(PrjFSKextLogDaemon_OSLogSubsystem, "daemon");
+    s_kextLogger = os_log_create(PrjFSKextLogDaemon_OSLogSubsystem, "kext");
+    
+    os_log(s_daemonLogger, "PrjFSKextLogDaemon starting up");
+
+    s_notificationPort = IONotificationPortCreate(kIOMasterPortDefault);
+    IONotificationPortSetDispatchQueue(s_notificationPort, dispatch_get_main_queue());
+
+    PrjFSService_WatchContext* watchContext = PrjFSService_WatchForServiceAndConnect(
+         s_notificationPort,
+         UserClientType_Log,
+         [](io_service_t service, io_connect_t connection, IOReturn connectResult, PrjFSService_WatchContext* context)
+         {
+             if (connectResult != kIOReturnSuccess || connection == IO_OBJECT_NULL)
+             {
+                io_string_t servicePath = "";
+                IORegistryEntryGetPath(service, kIOServicePlane, servicePath);
+                if (connectResult == kIOReturnUnsupported)
+                {
+                    CFTypeRef kextVersionObj = IORegistryEntryCreateCFProperty(service, CFSTR(PrjFSKextVersionKey), kCFAllocatorDefault, 0);
+                    os_log_error(
+                        s_daemonLogger,
+                        "Failed to connect to newly matched PrjFS kernel service at '%{public}s'; version mismatch. Expected %{public}s, kernel service version %{public}@",
```

Comment:
The idea was that `connectResult == kIOReturnUnsupported` would indicate a version mismatch, but I've just checked the `IOService` source, and there are actually some other cases where that error is raised on `IOServiceOpen()` so that's not ideal. I might just replace this with an extra boolean flag in the callback.